python-checks/src/main/java/org/sonar/python/checks/IdentityComparisonWithCachedTypesCheck.java

```diff
@@ -83,40 +114,24 @@ private static void checkIsComparison(SubscriptionContext ctx) {
     }
   }
 
-  private static boolean isComparisonToNone(IsExpression isExpr) {
-    return CheckUtils.isNone(isExpr.leftOperand().type()) || CheckUtils.isNone(isExpr.rightOperand().type());
+  private boolean isComparisonToNone(IsExpression isExpr) {
+    return isNone(isExpr.leftOperand().typeV2()) || isNone(isExpr.rightOperand().typeV2());
   }
 
-  /**
-   * Fully qualified names of constructors and functions that would are guaranteed to create fresh objects with
-   * references not shared anywhere else.
-   *
-   * If a reference arises from an call to one of these functions, and it does not escape anywhere else, then an
-   * <code>is</code>-comparison with such a reference will always return <code>False</code>.
-   *
-   * Note that these are fully qualified names of (value-level) expressions, not types.
-   */
-  private static final Set<String> FQNS_CONSTRUCTORS_RETURNING_UNIQUE_REF = new HashSet<>(
-    asList("frozenset", "bytes", "int", "float", "str", "tuple", "hash"));
-
-  /**
-   * Names of types that usually should not be compared with <code>is</code>.
-   *
-   * Note that these are names of types, not `fqn`s of expressions.
-   */
-  private static final Set<String> NAMES_OF_TYPES_UNSUITABLE_FOR_COMPARISON = new HashSet<>(
-    asList("frozenset", "bytes", "int", "float", "tuple"));
+  private boolean isNone(PythonType type) {
+    return isNoneTypeChecker.check(type) == TriBool.TRUE;
+  }
 
   /**
    * Checks whether an expression is either of a type that should generally be compared with <code>==</code>, or
    * whether it holds a non-escaping reference that stems from a function that guarantees that the
    * returned reference is not shared anywhere.
    */
-  private static boolean isUnsuitableOperand(Expression expr) {
-    InferredType tpe = expr.type();
-    if (isUnsuitableType(tpe)) {
-      if (expr.is(Tree.Kind.NAME)) {
-        Symbol symb = ((Name) expr).symbol();
+  private boolean isUnsuitableOperand(Expression expr) {
+    PythonType type = expr.typeV2();
+    if (isUnsuitableType(type)) {
+      if (expr instanceof Name name) {
+        SymbolV2 symb = name.symbolV2();
```

Comment:
Does the comment and condition below still make sense? I don't think `Name#symbolV2` can return null.